File: frontend/src/app/models/vocabulary.ts
export interface Vocabulary {
    word: string;
    context: {
        type: VocabularyType; 
        meaning: string
    }
}

export interface VocabularyType {
    displayName: string;
    value: string;
}

// export  VocabularyTypes = [];

//export type VocabularyType = 'Noun' | 'Verb' | 'Adverb' | 'Adjective' | 'Preposition';

File: backend/src/main/java/com/pph/vocabulary/model/Vocabulary.java
package com.pph.vocabulary.model;

public class Vocabulary {
	private String word;
	private VocabularyType context;
	
	public Vocabulary(String word, VocabularyType context) {
		this.word    = word;
		this.context = context;
	}

	public String getWord() {
		return word;
	}

	public void setWord(String word) {
		this.word = word;
	}

	public VocabularyType getContext() {
		return context;
	}

	public void setContext(VocabularyType context) {
		this.context = context;
	}
	
}
